doc/index.rst

```diff
@@ -1,8 +1,45 @@
-Plutus
-======
+Plutus Programming and Reference Guide
+=============================================
+
+About Plutus
+--------------------
+
+Plutus consists of Plutus Core, the programming, scripting and assembly language; 
+and Plutus Tx, the compiler that compiles the Haskell source code into Plutus Core 
```

Comment:
Plutus Tx usually refers to the compiler, I think. We can come up with some standard names for (1) the Haskell code compiled by Plutus Tx, which is a subset of Haskell; (2) the libraries we provide (e.g., PlutusTx.List) for writing such code.

Reply:
Yeah, we have always had some confusion here. Originally "Plutus Tx" mean "the subset of Haskell compiled by the plugin", but this is also confusing, since it's not really a language... the language is Haskell! So the terminology here has always been a mess.

And it's not helped by the fact that the `plutus-tx` package is now 90% standard library, rather than anything else.

Reply:
is there one compiler with multiple stages or multiple compilers? I hadn't heard this use of the term 'Plutus Tx' before. It might be worth mentioning Plutus IR as a signification intermediate language.

Reply:
@jmchapman I haven't seen a reference anywhere so far to Plutus IR. However, there are a lot of references in the Plutus Core readthedocs to the Plutus Tx compiler. Let me know, though, if you think this is an important distinction to make. I'd like to discuss further if needed.

Reply:
It would be good to sort all this out, yes. Maybe we need to talk about it, it's all bit murky and I'm not sure what the best thing to do is.

Reply:
> @jmchapman I haven't seen a reference anywhere so far to Plutus IR.

Right sorry, I am just looking at the PR without having looked at the rest of the documentation.

It's covered in this paper - https://iohk.io/en/research/library/papers/unraveling-recursion-compiling-an-ir-with-recursion-to-system-f/

Reply:
How about we schedule a meeting with the appropriate people sometime next week? I'll work on getting as clear as possible about what the precise questions are and send them out in advance.

Reply:
I'd like to get input from the team about how to reference Plutus IR. I've looked at the paper "Unraveling recursion: compiling an IR with recursion to System F" briefly. This would be one of the topics I'd like to put on our agenda for when we meet.